cm_modules/train.py

```diff
@@ -0,0 +1,122 @@
+# Copyright 2016 Goekcen Eraslan
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os
+import random
+
+from . import io
+# from .hyper import hyper
+
+import numpy as np
+import tensorflow as tf
+import keras.optimizers as opt
+from keras.callbacks import TensorBoard, ModelCheckpoint, EarlyStopping, ReduceLROnPlateau
+from keras import backend as K
+from keras.preprocessing.image import Iterator
+# The below lines were added due to error: Cannot convert keras symbolic input
+# to numpy array
+# Found this stack overflow solution: https://stackoverflow.com/questions/65366442/cannot-convert-a-symbolic-keras-input-output-to-a-numpy-array-typeerror-when-usi
+from tensorflow.python.framework.ops import disable_eager_execution
+
+disable_eager_execution()
+
+
+def train(adata, network, output_dir=None, optimizer='RMSprop', learning_rate=None,
+          epochs=300, reduce_lr=10, output_subset=None, use_raw_as_output=True,
+          early_stop=15, batch_size=32, clip_grad=5., save_weights=False,
+          validation_split=0.1, tensorboard=False, verbose=True, threads=None,
+          **kwds):
+
+    tf.compat.v1.keras.backend.set_session(
+        tf.compat.v1.Session(
+            config=tf.compat.v1.ConfigProto(
+                intra_op_parallelism_threads=threads,
+                inter_op_parallelism_threads=threads,
+            )
+        )
+    )
+    model = network.model
+    loss = network.loss
+
+    if output_dir is not None:
+        os.makedirs(output_dir, exist_ok=True)
+
+    if learning_rate is None:
```

Comment:
Do you still run into the error if you explicitly set the learning rate?

Reply:
I'm not sure if I understand what you mean by "setting the learning rate explicitly". If I do something like this `keras.optimizer.Adam(learning_rate=0.001)` I do still get the same error.

Reply:
Yeah that answers my question, thanks!